### ISSUE TITLE
Error while fetching app file group through GraphQL

### ISSUE THREAD

**Issue by Fen747** [bluelibs/bluelibs#163]
**Describe the bug**
Terminal error on server : 

<img width="1195" alt="Screen Shot 2022-01-27 at 15 50 19" src="https://user-images.githubusercontent.com/13072245/151383440-cc5ddb76-8256-43c4-9d2b-c08987ceb32f.png">

```
error  {
    "message": "value.forEach is not a function",
    "locations": [
        {
            "line": 2,
            "column": 3
        }
    ],
    "path": [
        "AppFileGroupsFind"
    ],
    "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
            "stacktrace": [
                "TypeError: value.forEach is not a function",
                "    at /Users/davidpanart/dev/selfcity/belett-v2/microservices/api/node_modules/@bluelibs/nova/src/core/query/hypernova/processDirectNode.ts:50:13",
                "    at Array.forEach (<anonymous>)",
                "    at processDirectNode (/Users/davidpanart/dev/selfcity/belett-v2/microservices/api/node_modules/@bluelibs/nova/src/core/query/hypernova/processDirectNode.ts:41:20)",
                "    at storeHypernovaResults (/Users/davidpanart/dev/selfcity/belett-v2/microservices/api/node_modules/@bluelibs/nova/src/core/query/hypernova/storeHypernovaResults.ts:31:22)",
                "    at processTicksAndRejections (internal/process/task_queues.js:95:5)",
                "    at async Promise.all (index 0)",
                "    at hypernovaRecursive (/Users/davidpanart/dev/selfcity/belett-v2/microservices/api/node_modules/@bluelibs/nova/src/core/query/hypernova/hypernova.ts:27:3)",
                "    at hypernova (/Users/davidpanart/dev/selfcity/belett-v2/microservices/api/node_modules/@bluelibs/nova/src/core/query/hypernova/hypernova.ts:33:3)",
                "    at AppFileGroupsCollection.queryGraphQL (/Users/davidpanart/dev/selfcity/belett-v2/microservices/api/node_modules/@bluelibs/mongo-bundle/src/models/Collection.ts:628:20)",
                "    at Object.AppFileGroupsFind (/Users/davidpanart/dev/selfcity/belett-v2/microservices/api/node_modules/@bluelibs/graphql-bundle/src/executor.ts:59:16)"
            ]
        }
    }
}
```

**To Reproduce**
1. Create an appFileGroup
2. Fetch it as shown above

**Comment by paulincai** [comment]
Chances are your 'value' is a Promise instead of an array. Are you using MongoDB 5 with the latest Node driver for Mongo? We've been seeing this in MeteorJS with the former Grapher package.

**Comment by theodorDiaconu** [comment]
@paulincai your theory is very valid. I asked for the data dump to try to replicate it locally, but why would you get a promise instead of an array?

**Comment by paulincai** [comment]
@theodorDiaconu check this update in Grapher (aggregate.js):

Before:
```
import { Promise } from 'meteor/promise';
Mongo.Collection.prototype.aggregate = function(pipelines, options = {}) {
    const coll = this.rawCollection();

    let result = Meteor.wrapAsync(coll.aggregate, coll)(pipelines, options);

    // We need to check If it's an AggregationCursor
    // The reason we do this was because of the upgrade to 1.7 which involved a mongodb driver update
    if (Array.isArray(result)) {
        return result;
    } else {
        return Promise.await(result.toArray());
    }
};
```

After:
```
import { Promise } from 'meteor/promise'

Mongo.Collection.prototype.aggregate = function (pipelines, options = {}) {
  return Promise.await(this.rawCollection().aggregate(pipelines).toArray())
}

```
In Meteor the issue is related to the rawCollection which uses the native driver (https://mongodb.github.io/node-mongodb-native/4.3/). Relevant migration instructions here: https://deploy-preview-11867--meteor-guide.netlify.app/2.6-migration.html (search for "rawCollection users" chapter)

**Comment by theodorDiaconu** [comment]
Understood @paulincai 

In this case there was a data inconsistency issue, there was a direct many relationship and instead of having an array of objectIds it was a string. Hence iterating with .forEach over a string failed.

Released @bluelibs/nova@1.3.5
